Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"NSMutableDictionary", @"class",
+                                                                                         refID, @"id",
+                                                                                         nil]];
+      GSXibElement *stringRecord   = [[GSXibElement alloc] initWithType: @"string"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"NS.key.0", @"key",
+                                                                                         nil]];
+      GSXibElement *placeholderRec = [[GSXibElement alloc] initWithType: @"object"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"IBUserDefinedRuntimeAttributesPlaceholder", @"class",
+                                                                                         @"IBUserDefinedRuntimeAttributesPlaceholderName", @"key",
+                                                                                         nil]];
+      GSXibElement *placeStringRec = [[GSXibElement alloc] initWithType: @"string"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"name", @"key",
+                                                                            nil]];
+      GSXibElement *referenceRec   = [self createReference: [element attributeForKey: @"id"]];
+
+      // Setup placeholder record...
+      [placeStringRec setValue: @"IBUserDefinedRuntimeAttributesPlaceholderName"];
+      [placeholderRec setElement: placeStringRec forKey: @"name"];
+      [placeholderRec setElement: referenceRec forKey: @"object"];
+      [placeholderRec setElement: element forKey: @"userDefinedRuntimeAttributes"];
+
+      // Attach placeholder and string records to the object record...
+      [objectRecord setElement: stringRecord forKey: @"NS.key.0"];
+      [objectRecord setElement: placeholderRec forKey: @"IBUserDefinedRuntimeAttributesPlaceholderName"];
+
+      // Add to flattened properties...
+      [_flattenedProperties setElement: objectRecord forKey: refID];
+
+      // Cleanup...
+      RELEASE(stringRecord);
+      RELEASE(placeholderRec);
+      RELEASE(placeStringRec);
+      RELEASE(objectRecord);
+    }
+}
+
+- (id) initForReadingWithData: (NSData*)data
+{
+  // Check data...
+  if (data == nil)
+    {
+      DESTROY(self);
+      return nil;
+    }
+#if     GNUSTEP_BASE_HAVE_LIBXML
```

Comment:
Why four spaces?

Reply:
I don't know. Please ask the person who added these in GSXibLoader.m the first time. Form there it was just copied over again and again. Looks like Richard did that four years ago right after our Dublin meeting. So all of us could have been involved...

Reply:
I removed these spaces in the new place.